pydantic/_internal/_core_utils.py

```diff
@@ -136,20 +136,13 @@ def _record_valid_refs(s: core_schema.CoreSchema, recurse: Recurse) -> core_sche
 
 def define_expected_missing_refs(
     schema: core_schema.CoreSchema, allowed_missing_refs: set[str]
-) -> core_schema.CoreSchema:
+) -> tuple[core_schema.CoreSchema, bool]:
```

Comment:
Potentially document or comment what the bool is here, I had to look at the callsite.

One option to avoid `bool` (and intermediate `tuple`, if you care about micro-optimizing) is to return `core_schema.CoreSchema | None` and return `None` in the case when there are no missing refs.

Reply:
Great call, done